### ISSUE TITLE
Unix domain socket in PostgreSQL URL

### ISSUE THREAD

**Issue by milgner** [hibernate/hibernate-reactive#1524]
For connecting to PostgreSQL instances via domain socket, the Vert.x documentation shows the example:

```java
PgConnectOptions connectOptions = new PgConnectOptions()
  .setHost("/var/run/postgresql")
  .setPort(5432)
  .setDatabase("the-db");
```

Unfortunately this doesn't work with Hibernate Reactive as the URI `postgres://%2Fvar%2Frun%2Fpostgresql/the-db` parses into the `host` field being `null`. 
My initial idea would be to have a PostgreSQL-specific check in the `DefaultSqlClientPoolConfiguration::connectOptions` method whether `host` is `null` and if that's the case and `authority` is filled in (which does contain the path to the socket), run it through URI decoding and use it as the host instead.

Not sure whether that is too specific, though?

**Comment by milgner** [comment]
I see that this particular part of the code has already undergone some major change in v2 alpha. I'll take a look and check whether I can get it to work with my application for testing purposes.

**Comment by milgner** [comment]
@DavideD thank you for that hint - that turned out to be a good solution! I added the service to the service registry and now everything works as expected! Setting up the dependencies for Nettys native transport and configuring Vert.x to prefer them turned out to be the actual fiddly part (and I still need to check whether it works for my colleagues with Apple M1 processors) - but that's another story.

While trying out Hibernate Reactive 2 alpha, I quickly had to abort my experiments as I found that it failed to use the `ClassLoaderService` which was was registered in the service registry. But I guess that might be just part of it being an alpha stage release.
